Add test for GitHub tokens, warning for password stored

setu4993 opened the pull request:
Resolves: https://github.com/python-poetry/poetry/issues/2348

Changes:
- Add tests for when `x-oauth-basic` GitHub URL is used in package link (https://github.com/python-poetry/poetry/issues/2348).
- Add warning and log for when a dependency stores a password in plain text by using the `is_unsafe` property (based on discussion in https://github.com/python-poetry/poetry/pull/2169#issuecomment-705587436).

<!-- This is just a reminder about the most common mistakes. Please make sure that you tick all *appropriate* boxes. But please read our [contribution guide](https://python-poetry.org/docs/contributing/) at least once, it will save you unnecessary review cycles! -->

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

@finswimmer : I have based this off of your branch that resovles https://github.com/python-poetry/poetry/issues/2062. Hopefully this helps in speeding things up towards getting that issue closed.

sdispater committed:
Make the Factory class more pluggable

abn committed:
Improve PEP-440 support for versions

This is a broad change aligning `poetry.core.semver.version.Version`
implementation to be closer to PEP-440 that to semver. This adds the
foundation for improved dev and local tag support in poetry managed
version as well as fix ordering bugs that existed due to the
differences between semantic versioning specification and PEP-440.

Further, previously used inline copy of `packaging.version.Version`
implementation has now been removed in favour of
`poetry.core.version.pep440.PEP440Version` implementation.

abn committed:
vendoring: split sync and update make targets

The sync target previously regenerated the poetry.lock for vendored
packages, causing larger than necessary change set for patches. This
change introduces the update target to specifically handle lock file
updates, lock target to regenerate the lock file and repurpose the sync
target to only sync as expected.

finswimmer committed:
fix (vcs.git): allow `+` in user name
new (vcs.git): extract user credential (passord, deployment key, ...) from git url
change (vcs.git): change order of init arguments for `ParsedUrl`
change (vcs.git): make user, password, port, name and rev optional for `ParsedUrl`

finswimmer committed:
change (vcs.git): make all arguments optional for `ParsedUrl`

finswimmer committed:
new (vcs.git): add property `is_unsafe` to `ParsedUrl` to return a boolean whether the ParsedUrl contains a secret like a password

hroncok committed:
Remove executable bits from Python files without shebangs

abn committed:
package: do not default to develop

Setting develop to `True` by default causes un-expected side-effects
and requires downstream consumers to rely on checks like
`package.develop and package.source_type == "directory""` to determine
if develop mode.

This also corresponds to current high-level behaviour of packages being
non-develop by default.

abn committed:
utils: fix canonical package name regex

Relates-to: python-poetry/poetry#3132
Relates-to: python-poetry/poetry#3200

abn committed:
Fix PEP 508 representation of dependency w/o extras

finswimmer committed:
fix incorrect type hints and add missing ones

roxchkplusony committed:
Ignore dev dependencies for PEP 517 builds

This change allows for development dependencies to be ignored creating
`Poetry` instances. This is used when PEP 517 artifacts are built as 
they are not required in this scenario.

Relates-to: python-poetry/poetry#2174

achow101 committed:
Make sdist deterministic by setting gzip mtime to 0

hroncok committed:
Extend the wheel Tag regex in tests to match cp3_10 and cp310

Fixes https://github.com/python-poetry/poetry-core/issues/110

abn committed:
actions: remove use of set-env command

Ref: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

setu4993 committed:
Add test for x-oauth github url

setu4993 committed:
Add warning, tests for when password is stored

ghost commented:
## DeepCode's analysis on [#15233b](https://www.deepcode.ai/app/gh/finswimmer/core/7c55cf74a7467fbdd547bc71f93635d4f69a7243/finswimmer/core/15233b31261a6517e7226b193035121f5c5eafb1/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=1&) found:

- :warning: **1** warning, :information_source: **1** minor issue. :point_down:
- :heavy_check_mark: **1** issue was fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">Combine filter then map into one comprehension. Occurrences: <ul><li><a href="https://github.com/finswimmer/core/blob/15233b31261a6517e7226b193035121f5c5eafb1/poetry/core/version/pep440/segments.py#L49">segments.py:49</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/finswimmer/core/7c55cf74a7467fbdd547bc71f93635d4f69a7243/finswimmer/core/15233b31261a6517e7226b193035121f5c5eafb1/pr/_/%2Fpoetry%2Fcore%2Fversion%2Fpep440%2Fsegments.py/python%2Fdc%2FcombineFilterAndMapIntoOneComp/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Use comprehensions instead of filter. Occurrences: <ul><li><a href="https://github.com/finswimmer/core/blob/15233b31261a6517e7226b193035121f5c5eafb1/poetry/core/version/pep440/segments.py#L51">segments.py:51</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/finswimmer/core/7c55cf74a7467fbdd547bc71f93635d4f69a7243/finswimmer/core/15233b31261a6517e7226b193035121f5c5eafb1/pr/_/%2Fpoetry%2Fcore%2Fversion%2Fpep440%2Fsegments.py/python%2Fdc%2FuseCompehensions/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/finswimmer/core/7c55cf74a7467fbdd547bc71f93635d4f69a7243/finswimmer/core/15233b31261a6517e7226b193035121f5c5eafb1/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=1&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=finswimmer)

setu4993 commented:
Fixed the deepcode CI issue, but not sure why the other tests are failing since it fails on the 'Install Poetry' step. Maybe the base branch needs to be rebased to the latest one from upstream?

I tested this locally and everything passes.

setu4993 commented on tests/vcs/test_vcs.py:
```diff
@@ -285,6 +302,17 @@ def test_normalize_url(url, normalized):
+                protocol="https",
+                resource="hostname",
+                pathname="/project/blah.git",
+                user="fafb334-cb038533f851c23d0b63254223Abf72ce4f02987e7064b0c95566699a",
+                password="x-oauth-basic",
```
This is not entirely correct since the `user` identified here is actually the personal access token, and the `password` is the auth protocol. However, it was the easiest thing to do without rewriting the regex matching and url parsing logic.

kasteph committed:
Default include formats to both sdist and wheel.

Commit:
Simplify temporary_directory and use safe_rmtree (#119)

* Simplify temporary_directory and use safe_rmtree

kurtmckee committed:
Demonstrate that the wheel builder does not close wheels explicitly

Related to python-poetry/poetry#3545

kurtmckee committed:
Explicitly close the wheel temp file after it's written

This resolves a crash that occurs on PyPy3 on Windows

Closes python-poetry/poetry#3545

audunska committed:
Fix #3258

Paths for generated code that's excluded by vcs, but then included
explicitly, were wrong because they appared in the exclude list during
Builder.find_excluded_files(). I changed find_excluded_files to not
exclude files which are explicitly included, by taking a set difference.

Added a test for this case.

audunska committed:
Black formatting

gi0baro committed:
Update packaging dep to v20.8

gi0baro committed:
Update GH test workflow

ShayNehmad-RecoLabs committed:
Add & to author name regex, resolve #3485 (#120)

Useful for R&D departments :) 
https://github.com/python-poetry/poetry/issues/3485

sdispater committed:
Merge pull request #125 from gi0baro/packaging-20-8

Update packaging dep to v20.8

sdispater committed:
Merge pull request #124 from cognitedata/fix-3285

Fix #3285

sdispater committed:
Merge pull request #122 from kurtmckee/fix-poetry-3545

Always close the wheel tempfile after writing to it

sdispater committed:
Merge pull request #105 from achow101/det-sdist

Make sdist deterministic by setting gzip mtime to 0

sdispater committed:
Merge branch 'master' into improvement/type-hinting

sdispater committed:
Merge pull request #104 from finswimmer/improvement/type-hinting

fix incorrect type hints and add missing ones

abn committed:
Ensure dependency string is PEP 508 compliant (#103)

finswimmer committed:
fix (core.version.helpers): add python 3.9.* to list of PYTHON_VERSION (#127)

sdispater committed:
Fix constraint upper bound including prereleases (#128)

Fixes python-poetry/poetry#2271

henryiii committed:
chore: packaging 20.9, Universal2 wheels on Apple Silicon (#129)

sdispater committed:
Revert changes to canonicalize_name (#132)

sdispater committed:
Update dependencies

sdispater committed:
Bump version to 1.0.1

sdispater committed:
Merge pull request #133 from python-poetry/release-1.0.1

Release 1.0.1

finswimmer committed:
"Dependency" needs to be imported outside TYPE_CHECKING only section (#134)

sdispater committed:
Update 1.0.1 change log

sdispater committed:
Bump version to 1.0.2

sdispater committed:
Merge pull request #135 from python-poetry/release-1.0.2

Release 1.0.2

sdispater committed:
Modernize code base (#131)

* Update CI files for Python 3.6+ only

* Modernize codebase

* Reorganize imports to improve performance

abn committed:
deps: update vendoring python constraint

abn committed:
Use deterministic time for generated sdist files

When generating setup.py and PKG-INFO files, ensure that generated
files use a deterministic timestamp to enhance reproducibility of
source distributions.

david-fortini committed:
Let poetry use default branch if none is specified. Resolves: python-poetry/poetry#3366

AndrewGuenther committed:
use uri for direct reference directory dependencies

abn committed:
tests: add PEP-440 comparison examples

abn committed:
pep440: add helper methods and fixes

abn committed:
pep440: fix bugs

abn committed:
Update pyproject to include compat

abn committed:
ci: expand python coverage for integration tests

abn committed:
ci: expand test matrix for integration tests

abn committed:
ci: add experimental support for python 3.10

abn committed:
ci: skip pypy for integration tests

abn committed:
tests: fix py3.6 compat issues in integration suite

abn committed:
package: add python 3.10 to available python list

abn committed:
update development dependencies

abn committed:
ci: switch to pip install when installing poetry

We do this since poetry releases currently do not work with python
3.10.

This can be replaced once new bootstrap is available.

abn committed:
pyproject: minor improvement to handling of data

abn committed:
schema: add support for source.links boolean

abn committed:
pep440: ensure epoch is retained on version bumps

abn committed:
Revert "Let poetry use default branch if none is specified. Resolves: python-poetry/poetry#3366"

This reverts commit 28b7a07a782327bc2bf09067ab2518ff0e5ea732.

abn committed:
ci: add downstreap pipeline

sdispater committed:
Update dependencies

sdispater committed:
Bump version to 1.1.0a1

abn committed:
lark: lazy load parsers (#150)

sdispater committed:
Merge pull request #151 from python-poetry/release-1.1.0a1

Release 1.1.0a1

abn committed:
pep440: replace lark in favour of regex  (#152)

* tests: fix incorrect parameters

* pep440: replace lark in favour of regex

The lark early implementation of PEP440 that was introduced was not
performant. The implementation added on average around 7 seconds to
the test suite execution time.

This change drops the lark parser, and uses `packaging.version`
provided regex for version string parsing. New dataclass interface for
version instances remain unaffected.

sdispater committed:
Bump version to 1.1.0a2

sdispater committed:
Fix an error when handling single digit Python markers

abn committed:
pep440: support upper bound post/local release comparisons (#157)

* tests: add coverage for poetry.core.version.pep440

* pep440: allow release tuples

* pep440: support post/local release comparisons

This change ensures that post and local releases are taken into
consideration when checking if version range allows a post release
local build release at upper and lower bounds.

The following conditions now hold for upper bound checks.

- `<=3.0.0` allows `3.0.0+local.1`, `3.0.0-1`
- `<=3.0.0+local.1` disallows `3.0.0+local.2`, allows `3.0.0-1`
- `<=3.0.0-1` allows `3.0.0+local.1`, `3.0.0`

Lower bound checks require no modification and works due to the
implicit version comparison of `poetry.core.pep440.PEP440Version`.

abn committed:
pep440: support post/local for semver allows (#158)

This change ensures that post and local releases are taken into
consideration when checking if semver version instance allows
post and local build releases.

The following conditions now hold `poetry.core.semver.Version.allows`.

- `3.0.0` allows `3.0.0+local.1`, `3.0.0-1`
- `3.0.0+local.1` disallows `3.0.0+local.2`, allows `3.0.0-1`
- `3.0.0-1` disallows ``3.0.0`, `3.0.0+local.1`, allows `3.0.0-1+local.1`

sdispater committed:
Merge pull request #161 from python-poetry/release-1.1.0a2

Release 1.1.0a2

sdispater committed:
Fix dependency markers not being copied in with_constraint()

abn committed:
deps: relax compatibility requirement constraints (#164)

sdispater committed:
Bump version to 1.1.0a3

sdispater committed:
Merge pull request #166 from python-poetry/release-1.1.0a3

Release 1.1.0a3

kasteph committed:
pep440: preserve leading zeros in local segment

Resolves: python-poetry/poetry#3705

abn committed:
ci: update test scripts to use install-poetry.py

This change also fixes python 3.10 failures.

abn committed:
ci: cleanup release workflow

abn committed:
pre-commit: add json checks

Commit:
fix(packages/dependency): add space after filename for file dependencies with markers (#153)

* fix(packages/dependency): add space after filename for file dependencies with markers

local vendored files need a space after the file name and before the ";" which demarks the start of
markers

fix #3872

setu4993 committed:
Set optional args to ParsedUrl to None as default

setu4993 committed:
Add missing format vars for ParsedUrl

setu4993 committed:
Apply black
